src/calendar-heatmap.tsx

```diff
@@ -0,0 +1,1735 @@
+      .style('opacity', 0)
+      .remove();
+    this.labels.selectAll('.label-day').remove();
+    this.labels.selectAll('.label-month').remove();
+    this.hideBackButton();
+  }
+
+  /**
+   * Transition and remove items and labels related to month overview
+   */
+  removeMonthOverview() {
+    this.items
+      .selectAll('.item-block-month')
+      .selectAll('.item-block-rect')
+      .transition()
+      .duration(this.state.settings.transition_duration)
+      .ease(easeLinear)
+      .style('opacity', 0)
+      .attr('x', (_d, i) => {
+        return i % 2 === 0
+          ? -this.state.settings.width / 3
+          : this.state.settings.width / 3;
+      })
+      .remove();
+    this.labels.selectAll('.label-day').remove();
+    this.labels.selectAll('.label-week').remove();
+    this.hideBackButton();
+  }
+
+  /**
+   * Transition and remove items and labels related to week overview
+   */
+  removeWeekOverview() {
+    this.items
+      .selectAll('.item-block-week')
+      .selectAll('.item-block-rect')
+      .transition()
+      .duration(this.state.settings.transition_duration)
+      .ease(easeLinear)
+      .style('opacity', 0)
+      .attr('x', (_d, i) => {
+        return i % 2 === 0
+          ? -this.state.settings.width / 3
+          : this.state.settings.width / 3;
+      })
+      .remove();
+    this.labels.selectAll('.label-day').remove();
+    this.labels.selectAll('.label-week').remove();
+    this.hideBackButton();
+  }
+
+  /**
+   * Transition and remove items and labels related to daily overview
+   */
+  removeDayOverview() {
+    this.items
+      .selectAll('.item-block')
+      .transition()
+      .duration(this.state.settings.transition_duration)
+      .ease(easeLinear)
+      .style('opacity', 0)
+      .attr('x', (_d, i) => {
+        return i % 2 === 0
+          ? -this.state.settings.width / 3
+          : this.state.settings.width / 3;
+      })
+      .remove();
+    this.labels.selectAll('.label-time').remove();
+    this.labels.selectAll('.label-project').remove();
+    this.hideBackButton();
+  }
+
+  /**
+   * Helper function to hide the back button
+   */
+  hideBackButton() {
+    this.buttons
+      .selectAll('.button')
+      .transition()
+      .duration(this.state.settings.transition_duration)
+      .ease(easeLinear)
+      .style('opacity', 0)
+      .remove();
+  }
+
+  constructor(props: CalendarHeatmapProps) {
+    super(props);
+
+    this.state = {
+      settings: {
+        gutter: 5,
+        item_gutter: 1,
+        width: 1000,
+        height: 200,
+        item_size: 10,
+        label_padding: 40,
+        max_block_height: 20,
+        transition_duration: 500,
+        tooltip_width: 250,
+        tooltip_padding: 15,
+      },
+      in_transition: false,
+      // First entry for history should be 'global', so that even if the user enters the chart first time he will be able to go back till global overview
+      history: ['global', props.overview ?? 'year'],
```

Comment:
@qur786 refactor such that you do this only when `props.overivew` is not `global`